File: iosApp/iosApp/Presentation/Navigation/NavigationAction.swift
//
//  NavigationAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/05.
//

import Foundation

enum NavigationAction {
    case navigateToMain(initialTab: String? = nil)
    case navigateToCast(id: String)
    case navigateToCafe(id: String)
    case navigateToCafeEvent(cafeId: String, eventId: String, showCafeButton: Bool = false)
    case replaceWithCafe(id: String)
    case navigateToCafeDashboard(id: String)
    case navigateToBanner(cafeId: String? = nil)
    case navigateToBannerEdit(cafeId: String? = nil, bannerId: String? = nil)
    case navigateToExternalLink(title: String, url: String)
    case navigateToCafeInfoEdit(id: String?, isRegistrationMode: Bool = false)
    case navigateToNoticeEvent(id: String)
    case navigateToCastEdit(cafeId: String?, castId: String?)
    case navigateToSchedule(castId: String? = nil)
    case navigateToCastManagement(cafeId: String, cafeName: String)
    case navigateToCastList(cafeId: String)
    case navigateToMenuGoods(id: String)
    case navigateToMenuGoodsEdit(cafeId: String, itemId: String? = nil)
    case navigateToReviewEdit(cafeId: String, reviewId: String? = nil)
    case navigateToPicture(imageUrl: String)
    case navigateToCheckInMap(initialRegionKey: String? = nil)
    case navigateToSignIn
    case navigateToSignUp
    case navigateToResetPassword
    case navigateToNotification
    case navigateToSettings
    case navigateToNotificationSettings
    case navigateToAccountSettings
    case navigateToInquiry
    case navigateToUserManagement
    case navigateToDormantAccount
    case navigateToChangePassword
    case navigateToCommunity
    case navigateToPostEdit(postId: String? = nil)
    case navigateToPostDetail(postId: String)
    case navigateBack
    case refreshUnreadNotificationCount
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/navigation/NavigationAction.kt
package com.hhp227.concafe.presentation.navigation

sealed interface NavigationAction {
    data class NavigateToMain(val initialTab: String? = null) : NavigationAction
    data class NavigateToCast(val id: String) : NavigationAction
    data class NavigateToCafe(val id: String) : NavigationAction
    data class NavigateToCafeEvent(val cafeId: String, val eventId: String) : NavigationAction
    data class NavigateToCafeDashboard(val id: String) : NavigationAction
    data class NavigateToBanner(val cafeId: String? = null) : NavigationAction
    data class NavigateToBannerEdit(
        val cafeId: String? = null,
        val bannerId: String? = null
    ) : NavigationAction
    data class NavigateToExternalLink(val title: String, val url: String) : NavigationAction
    data class NavigateToCafeInfoEdit(
        val id: String? = null,
        val isRegistrationMode: Boolean = false
    ) : NavigationAction
    data class NavigateToNoticeEvent(val id: String) : NavigationAction
    data class NavigateToCastEdit(val cafeId: String? = null, val castId: String? = null) : NavigationAction
    data class NavigateToSchedule(val castId: String? = null) : NavigationAction
    data class NavigateToCastManagement(val cafeId: String, val cafeName: String) : NavigationAction
    data class NavigateToMenuGoods(val id: String) : NavigationAction
    data class NavigateToMenuGoodsEdit(val cafeId: String, val itemId: String? = null) : NavigationAction
    data class NavigateToReviewEdit(val cafeId: String, val reviewId: String? = null) : NavigationAction
    data class NavigateToPicture(val imageUrl: String) : NavigationAction
    data class NavigateToCheckInMap(val initialRegionKey: String? = null) : NavigationAction
    data object NavigateToSignIn : NavigationAction
    data object NavigateToSignUp : NavigationAction
    data object NavigateToResetPassword : NavigationAction
    data object NavigateToNotification : NavigationAction
    data object NavigateToSettings : NavigationAction
    data object NavigateToNotificationSettings : NavigationAction
    data object NavigateToAccountSettings : NavigationAction
    data object NavigateToInquiry : NavigationAction
    data object NavigateToChangePassword : NavigationAction
    data object NavigateToCommunity : NavigationAction
    data class NavigateToPostEdit(val postId: String? = null) : NavigationAction
    data class NavigateToPostDetail(val postId: String) : NavigationAction
    data object NavigateBack : NavigationAction
    data object RefreshUnreadNotificationCount : NavigationAction
}
